test/Serialization/Recovery/private-conformances.swift

```diff
@@ -0,0 +1,34 @@
+// RUN: %empty-directory(%t)
+
+// Build the protocol library and conforming library to include PrivateProtocol.
+// RUN: %target-swift-frontend -emit-module -DPROTOCOL_LIB -DBEFORE %s -module-name protocol_lib -emit-module-path %t/protocol_lib.swiftmodule
+// RUN: %target-swift-frontend -emit-module -DCONFORMS_LIB %s -module-name conforms_lib -emit-module-path %t/conforms_lib.swiftmodule -I %t
+
+// Rebuild the protocol library to omit PrivateProtocol.
+// RUN: %target-swift-frontend -emit-module -DPROTOCOL_LIB -DAFTER %s -module-name protocol_lib -emit-module-path %t/protocol_lib.swiftmodule
+
+// The conforming library's types should still deserialize.
+// RUN: %target-swift-frontend -emit-module -DCLIENT %s -module-name client -emit-module-path %t/client.swiftmodule -I %t
+
+#if PROTOCOL_LIB
+
+#if BEFORE
+public protocol PrivateProtocol {}
+#endif
+
+#elseif CONFORMS_LIB
+import protocol_lib
+
+public protocol PublicProtocol {}
+
+public struct S : PrivateProtocol, PublicProtocol {}
+public enum E : PrivateProtocol, PublicProtocol {}
+public class C : PrivateProtocol, PublicProtocol {}
```

Comment:
Needs an extension test case that doesn't involve nested types. I wonder if just asking `swift-ide-test` to print the module will have the right balance of "deserialize all the things, but don't do any lookups".